### PR TITLE
#616 added modal on icon click

### PR DESCRIPTION
**Pull request by jinapurapu**
Upon clicking the pencil icon, a modal opens with fields for current password, new password, and to reenter the new password. The modal is largely a copy of the ChangePassword modal, and requires further modification to pass the selected user, and change the selected users password.

Screenshot from 2021-05-14 18-48-48
Screenshot from 2021-05-14 18-49-26

 fixes https://github.com/miniohq/engineering/issues/3

### CODE REVIEW

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        noValidate
+        autoComplete="off"
+        onSubmit={(e: React.FormEvent<HTMLFormElement>) => {
+          changePassword(e);
+        }}
+      >
+        <Grid container>
+          <Grid item xs={12} className={classes.formScrollable}>
+            <Grid item xs={12}>
+              <InputBoxWrapper
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:125]: When an admin is changing someone else's password, the current password is not needed, so we can remove this field

File: portal-ui/src/screens/Console/Buckets/types.tsx [hunk 2]
```diff
@@ -93,6 +93,7 @@ export interface MakeBucketRequest {
 }
 
 export interface ChangePasswordRequest {
+  //selected_user: object;
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Buckets/types.tsx:96]: if this line is not needed, remove it

File: restapi/user_account.go [hunk 3]
```diff
@@ -55,6 +55,18 @@ func changePassword(ctx context.Context, client MinioAdmin, session *models.Prin
 	return nil
 }
 
+// changeUserPassword validate current current user password and if it's correct set the new password
+// WHAT SHOULD IT DO? just set the new password, assuming this is for locked out users, current password may be unknown, only Admin users can use this function, should it use 2factor/email?
+func changeUserPassword(ctx context.Context, client MinioAdmin, session *models.Principal, currentSecretKey, newSecretKey string) error {
+	if session.AccountSecretKey != currentSecretKey {
```
Comment (dvaldivia) [inline restapi/user_account.go:61]: we probably don't have the user's current password, so we shouldn't require this field

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: bindata.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (dvaldivia) [inline bindata.go:1]: please remove this file

Comment (dvaldivia) [reply bindata.go:1]: any bindata file mnust be removed from this PR

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: restapi/user_account.go [hunk 5]
```diff
@@ -55,6 +55,18 @@ func changePassword(ctx context.Context, client MinioAdmin, session *models.Prin
...
+		return errChangePassword
+	}
+	if err := client.changePassword(ctx, session.AccountAccessKey, newSecretKey); err != nil {
```
Comment (dvaldivia) [inline restapi/user_account.go:64]: do use `session.AccountAccessKey`

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    },
+    ...actionsTray,
+    ...modalBasic,
+    ...containerForHeader(theme.spacing(4)),
+  });
+
+interface IChangePasswordProps {
+  classes: any;
+  open: boolean;
+  // selectedUser: User | null;
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:47]: We can remove this line

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+  closeModal: () => void;
+  setModalErrorSnackMessage: typeof setModalErrorSnackMessage;
+}
+
+const ChangePassword = ({
+  classes,
+  open,
+  //selectedUser,
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:55]: We can remove this line if not needed

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  const changePassword = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (loading) {
+      return;
+    }
+    setLoading(true);
+
+    let request: ChangeUserPasswordRequest = {
+      //selected_user: selectedUser,
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:72]: We can remove this line

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+      new_secret_key: newPassword,
+    };
+
+    api
+      .invoke("POST", "/api/v1/account/change-user-password", request)
+      .then((res) => {
+        setLoading(false);
+        // setSelectedUser("");
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:80]: We can remove this line

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+          <Grid item xs={12} className={classes.buttonContainer}>
+            <Button
+              type="submit"
+              variant="contained"
+              color="primary"
+              disabled={
+                loading ||
+                !(
+                  newPassword.length > 0 &&
+                  reNewPassword.length > 0
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:148]: we can add another condition to disable the button if passwords don't match
```suggestion
                  reNewPassword.length > 0 && newPassword !== reNewPassword
```

File: portal-ui/src/screens/Console/Groups/Groups.tsx [hunk 11]
```diff
@@ -189,7 +189,7 @@ const Groups = ({ classes, setErrorSnackMessage }: IGroupsProps) => {
         <Grid item xs={12} className={classes.container}>
           <Grid item xs={12} className={classes.actionsTray}>
             <TextField
-              placeholder="Search Groups"
+              placeholder="Search "
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Groups/Groups.tsx:192]: I think this label should not be removed

File: swagger.yml [hunk 12]
```diff
@@ -79,7 +79,7 @@ paths:
           description: Generic error response.
           schema:
             $ref: "#/definitions/error"
-      security: [ ]
+      security: []
```
Comment (bexsoft) [inline swagger.yml:82]: I think you need to update your prettier too, if you installed it using brew please run this command in a terminal:
`brew upgrade prettier`

The version that we are using right now is `2.3.0`

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Comment by bexsoft** [comment]
Please review:
```
restapi/admin_users.go:109:2: missing ',' before newline in argument list
restapi/admin_users.go:111:6: expected '(', found listUsers
```

File: portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx (hunk 1 above)
Comment (jinapurapu) [reply portal-ui/src/screens/Console/Account/ChangeUserPasswordModal.tsx:114]: done

File: restapi/admin_users.go [hunk 13]
```diff
@@ -540,3 +548,48 @@ func getListUsersWithAccessToBucketResponse(session *models.Principal, bucket st
...
+	user := *params.Body.SelectedUser
+	newSecretKey := *params.Body.NewSecretKey
+
+	// changes password of user to newSecretKey
+	if err := changeUserPassword(ctx, adminClient, user, newSecretKey); err != nil {
+		return prepareError(err)
+	}
+	// user credentials are updated at this point, we need to generate a new admin client and authenticate using
+	// the new credentials
+	//credentials, err := getConsoleCredentials(ctx, accessKey, newSecretKey)
```
Comment (dvaldivia) [inline restapi/admin_users.go:581]: remove commented code if not needed

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — APPROVED** [review]
Tested Works

**Review by bexsoft — APPROVED** [review]

**Pull request merged**